clang/test/SemaSYCL/esimd_widen.cpp

```diff
@@ -0,0 +1,28 @@
+// RUN: %clang_cc1 -fsycl-is-device -fsyntax-only -verify %s
+
+// This test check the semantics of sycl_esimd_widen attribute
+
+// expected-error@+1{{'sycl_esimd_widen' attribute argument must be 8, 16, or 32}}
+__attribute__((sycl_esimd_widen(17))) void foo1() {}
```

Comment:
Can you also add a test that shows we diagnose when you try to apply the attribute to a non-function or is given a non-integer argument?

Reply:
Done